.github/workflows/ci.yml

```diff
@@ -119,85 +99,89 @@ jobs:
         rust-version: ${{ matrix.toolchain.rust }}
         targets: ${{ matrix.platform.target }}
 
-    - name: Install cargo-ndk
-      if: contains(matrix.platform.target, 'android')
-      run: cargo install --version '<2.0.0' cargo-ndk
+    - name: Rust cache
+      uses: Swatinem/rust-cache@v1
```

Comment:
Should this be given a key with the runner image name and matrix target, or can it sniff those out automatically?

Reply:
This setup will probably be ok until we start the cross builds once more (still not convinced we should do those). Just thinking since it's part of a matrix build, perhaps it should be keyed manually. It'd seem that ubuntu and lint-rust differ only on the the prefix:

- v0-rust-lint-rust-1.54.0-x86_64-unknown-linux-gnu-a178d0322ce2-5bae624ed82b92875943
- v0-rust-ci-matrix-1.54.0-x86_64-unknown-linux-gnu-a178d0322ce2-5bae624ed82b92875943

If we'd use ubuntu to do the cross compilation into android, it would get the same cache keys.

Reply:
It uses a smart enough combination:
```
This cache is automatically keyed by:

    the github job,
    the rustc release / host / hash, and
    a hash of the Cargo.lock / Cargo.toml files (if present).
    a hash of the rust-toolchain / rust-toolchain.toml files (if present).
```

Reply:
If we'd enable the cross builds once more, as senseless it would seem, it would collide on ubuntu (and maybe macos), because doesn't include the target. Both ubuntu and android runs would produce competing caches as result, probably flipflopping between which gets cached.

Reply:
I was assuming that each matrix target would get its own job, but I see now that 
```
v0-rust-ci-matrix-1.54.0-x86_64-unknown-linux-gnu-a178d0322ce2-5bae624ed82b92875943
```
this implies that won't be the case. I'll add the target to the key.

Reply:
I've added `matrix.platform.target` to the cache key. Note that `key` extends the automatic key the action uses (i.e. not replace).